evdns.c

```diff
@@ -742,9 +742,32 @@ nameserver_failed(struct nameserver *const ns, const char *msg) {
 	ns->state = 0;
 	ns->failed_times = 1;
 
-	disconnect_and_free_connection(ns->connection);
-	ns->connection = NULL;
+	if (ns->connection) {
+		disconnect_and_free_connection(ns->connection);
+		ns->connection = NULL;
+	} else if (err == ENOTCONN) {
+		// XXX: https://github.com/libevent/libevent/issues/265
```

Comment:
And please can you describe the problem in the comments, providing link is not enough

Reply:
fixed